Common/Brokerages/AlpacaBrokerageModel.cs

```diff
@@ -0,0 +1,133 @@
+    {
+        /// <summary>
+        /// The default markets for the alpaca brokerage
+        /// </summary>
+        public new static readonly IReadOnlyDictionary<SecurityType, string> DefaultMarketMap = new Dictionary<SecurityType, string>
+        {
+            {SecurityType.Base, Market.USA},
+            {SecurityType.Equity, Market.USA}
+        }.ToReadOnlyDictionary();
+
+        /// <summary>
+        /// Gets a map of the default markets to be used for each security type
+        /// </summary>
+        public override IReadOnlyDictionary<SecurityType, string> DefaultMarkets => DefaultMarketMap;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DefaultBrokerageModel"/> class
+        /// </summary>
+        /// <param name="accountType">The type of account to be modelled, defaults to
+        /// <see cref="AccountType.Margin"/></param>
+        public AlpacaBrokerageModel(AccountType accountType = AccountType.Margin)
+            : base(accountType)
+        {
+        }
+
+        /// <summary>
+        /// Returns true if the brokerage could accept this order. This takes into account
+        /// order type, security type, and order size limits.
+        /// </summary>
+        /// <remarks>
+        /// For example, a brokerage may have no connectivity at certain times, or an order rate/size limit
+        /// </remarks>
+        /// <param name="security"></param>
+        /// <param name="order">The order to be processed</param>
+        /// <param name="message">If this function returns false, a brokerage message detailing why the order may not be submitted</param>
+        /// <returns>True if the brokerage could process the order, false otherwise</returns>
+        public override bool CanSubmitOrder(Security security, Order order, out BrokerageMessageEvent message)
+        {
+            message = null;
+
+            // validate security type
+            if (security.Type != SecurityType.Equity)
+            {
+                message = new BrokerageMessageEvent(BrokerageMessageType.Warning, "NotSupported",
+                    $"The {nameof(AlpacaBrokerageModel)} does not support {security.Type} security type."
+                );
+
+                return false;
+            }
+
+            // validate order type
+            if (order.Type != OrderType.Limit && order.Type != OrderType.Market && order.Type != OrderType.StopMarket && order.Type != OrderType.StopLimit && order.Type != OrderType.MarketOnOpen)
+            {
+                message = new BrokerageMessageEvent(BrokerageMessageType.Warning, "NotSupported",
+                    $"The {nameof(AlpacaBrokerageModel)} does not support {order.Type} order type."
+                );
+
+                return false;
+            }
+
+            // validate time in force
+            if (order.TimeInForce != TimeInForce.GoodTilCanceled && order.TimeInForce != TimeInForce.Day)
```

Comment:
I'm not sure these equality checks will work as expected, perhaps type checks are intended? Accompanying unit tests for the brokerage model would help to alleviate this concern.

Reply:
Well, the validation of every parameter is needed in any brokerage, I think.

And I saw it on every brokerage here in QC.

Reply:
Those models may have bugs too :/

Consider the following failing unit test:

``` csharp
[Test]
public void NewDayTimeInForceEqualsStaticTimeInForceDay()
{
    var order = new MarketOrder();
    order.Properties.TimeInForce = new DayTimeInForce();
    Assert.AreEqual(order.TimeInForce, TimeInForce.Day);
}
```